### PR TITLE
feat(dlp): added a sample for dlp reidentify fpe

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp reidentify fpe](https://cloud.google.com/dlp/docs/samples/dlp-reidentify-text-fpe) with a unit test case.
- Apart from that updated the deid_fpe.go, The key decoding is not proper in that code so updated it.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/deid_fpe.go [hunk 1]
```diff
@@ -49,11 +49,13 @@ func deidentifyFPE(w io.Writer, projectID, input string, infoTypeNames []string,
 	for _, it := range infoTypeNames {
 		infoTypes = append(infoTypes, &dlppb.InfoType{Name: it})
 	}
+
 	// Read the key file.
```
Comment (annadata) [inline dlp/snippets/deid/deid_fpe.go:53]: Is this comment accurate now?

File: dlp/snippets/deid/deid_fpe.go [hunk 2]
```diff
@@ -72,8 +74,8 @@ func deidentifyFPE(w io.Writer, projectID, input string, infoTypeNames []string,
 										CryptoKey: &dlppb.CryptoKey{
 											Source: &dlppb.CryptoKey_KmsWrapped{
 												KmsWrapped: &dlppb.KmsWrappedCryptoKey{
-													WrappedKey:    keyBytes,
-													CryptoKeyName: cryptoKeyName,
+													WrappedKey:    kmsWrappedCryptoKey,
+													CryptoKeyName: keyFileName,
```
Comment (annadata) [inline dlp/snippets/deid/deid_fpe.go:78]: The changes in this file are confusing for me. 
Are you moving from using file contents to kmsWrappedKey from env vars?
If yes, why is CryptoKeyName using keyFileName?

File: dlp/snippets/deid/deid_fpe.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_fpe.go:53]: Updated the comment.

File: dlp/snippets/deid/deid_fpe.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_fpe.go:78]: Thank you @annadata  for your feedback, Actually, the code for the decoding key is not working properly, so the KeyFIleReading part only I have changed, also not taking it as an environment variable.

**Review by telpirion — APPROVED** [review]

File: dlp/snippets/deid/deid_fpe.go (hunk 2 above)
Comment (annadata) [reply dlp/snippets/deid/deid_fpe.go]: @AarshD-crest Can you please help fix the naming that is confusing?

Comment (AarshD-crest) [reply dlp/snippets/deid/deid_fpe.go:78]: @annadata The variable names in `reid_text_fpe.go` is intuitive and not confusing, But I have Updated the variable names in `deid_fpe.go` as it sounds confusing.

**Pull request merged**